src/firstpass.rs

```diff
@@ -1183,6 +1235,55 @@ impl<'a, 'b> FirstPass<'a, 'b> {
             None
         }
     }
+
+    /// Parses an attribute block content, such as `.class1 #id .class2`.
+    ///
+    /// Returns `(id, classes)`.
+    ///
+    /// It is callers' responsibility to find opening and closing characters of the attribute
+    /// block.
+    ///
+    /// If `last_ix` is less than `first_index`, attribute block is considered as empty.
+    ///
+    /// Note that this parsing requires explicit whitespace separators between
+    /// attributes. This is intentional design with the reasons below:
+    ///
+    /// * to keep conversion simple and easy to understand for any possible input,
+    /// * to avoid adding less obvious conversion rule that can reduce compatibility
+    ///   with other implementations more, and
+    /// * to follow the major design of implementations with the support for the
+    ///   attribute blocks extension (as of this writing).
+    ///
+    /// See also: [`Options::ENABLE_HEADING_ATTRIBUTES`].
+    ///
+    /// [`Options::ENABLE_HEADING_ATTRIBUTES`]: `crate::Options::ENABLE_HEADING_ATTRIBUTES`
+    fn parse_inside_attribute_block(
+        &self,
+        first_ix: usize,
+        last_ix: usize,
+        id: &mut Option<CowStr<'a>>,
```

Comment:
It seems the `CowStr` is only ever a `Borrowed` variant for both `id` and `classes`; in that case we could just use `&'a str`.

Reply:
Also, we're not setting `id` and `classes` anywhere else, are we? It may be preferable to not pass mutable refs, but rather a `(Option<&'a str>, Vec<&'a str>)` instead.

Reply:
Thank you, returning `(Option<_>, Vec<_>)` seems better. I'll rewrite to that style.

I used `CowStr<'a>` instead of `&'a str` here since I think it could be (theoretically) possible to accept some kind of escape in some future.
(For example, `.foo\}.bar` into `class="foo}bar"` or `.α` into `class="%CE%B1"`, with more extension options...?)

I'm not sure it can be when and how, or whether it will really happen or not.
For now it seems such conversions won't happen in near future, as there are no standards or widely-accepted rules.
However, I'm not confident that the ID and classes continue to be unmodified string, so I used `CowStr<'a>` to avoid possible future breaking change.

At this time it seems OK to use `&'a str` and `Vec<&'a str>`, but it can be read as a message that says the ID and classes are guaranteed to be unmodified, and I'm not sure it is bad or not...
How do you think?